Essentials/src/main/java/com/earth2me/essentials/BalanceTopImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.earth2me.essentials;
+
+import net.ess3.api.IEssentials;
+import net.essentialsx.api.v2.services.BalanceTop;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public class BalanceTopImpl implements BalanceTop {
+    private final IEssentials ess;
+    private List<Map.Entry<String, BigDecimal>> topCache = new ArrayList<>();
```

Comment:
Guava anyone ?

Reply:
absolutely not

Reply:
don't need any of guava's cache features. the whole list is thrown out periodically anyway.